identity/src/main/java/com/android/identity/Util.java

```diff
@@ -298,15 +298,39 @@ static boolean cborDecodeBoolean(@NonNull byte[] data) {
         return simple.getSimpleValueType() == SimpleValueType.TRUE;
     }
 
+    /**
+     * Accepts a {@code DataItem}, attempts to cast it to a {@code Number}, then returns the value
+     * Throws {@code IllegalArgumentException} if the {@code DataItem} is not a {@code Number}. This
+     * method also checks bounds, and if the given data item is too large to fit in a long, it
+     * throws {@code ArithmeticException}.
+     */
+    static long asLong(DataItem item) {
```

Comment:
Naming: checkedLongValue() feels more idiomatic?

Maybe this method should take (or delegate to a method that takes) a BigInteger instead of a DataItem? WDYT? E.g.

~~~
static long checkedLongValue(DataItem item) {
  return checkedLongValue(castTo(Number.class, item).getValue());
}

static long checkedLongValue(BigInteger bigInteger) {
  ...
~~~